src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,46 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+
     private final Executor executor;
 
+    private final Map<String, UserData> cache;
+
+    private final Random random;
+
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
+        this.random = new Random();
+        this.cache = new ConcurrentHashMap<>();
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> fetchUserDetail(userId), executor);
+    }
+
+    public UserData fetchUserDetail(String userId) {
+        try {
+            Thread.sleep(random.nextInt(100));
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
The `InterruptedException` should be handled more gracefully. Instead of throwing a `RuntimeException`, consider restoring the interrupt status by calling `Thread.currentThread().interrupt()` and then handling the exception appropriately.